### PR TITLE
chore: fix `npm run perf` crashes

### PR DESCRIPTION
**Pull request by aladdin-add**
it was crashing as eslint treat the `.eslintrc.yml` to be js.
this commit changes the generated configs to `eslint.config.js`.

refs: https://github.com/eslint/eslint/issues/16255#issuecomment-1229817918

#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

### CODE REVIEW

**Review by nzakas — APPROVED** [review]
LGTM. Would like another set of eyes to double check before merging.

**Comment by mdjermanovic** [comment]
I'm getting `No files matching the pattern "tmp\eslint\performance\eslint\{lib,tests\lib}\**\*.js" were found.` on Windows, though that seems to be because of https://github.com/eslint/eslint/issues/16259.

File: Makefile.js
Comment (mdjermanovic) [inline Makefile.js:86]:
```suggestion
    PERF_MULTIFILES_TARGETS = `"${TEMP_DIR}eslint/performance/eslint/{lib,tests/lib}/**/*.js"`,
```

There is an extra `/` because `TEMP_DIR` already ends with a `/`.

**Pull request merged**
